checker/src/org/checkerframework/checker/lock/wholefiles/LockAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,261 @@
+                // Two @GuardedBy annotations are considered subtypes of each other if and only if their values match exactly.
+
+                List<String> lhsValues =
+                    AnnotationUtils.getElementValueArray(lhs, "value", String.class, true);
+                List<String> rhsValues =
+                    AnnotationUtils.getElementValueArray(rhs, "value", String.class, true);
+
+                return rhsValues.containsAll(lhsValues) && lhsValues.containsAll(rhsValues);
+            }
+
+            // Remove values from @GuardedBy annotations (and use the Checker Framework's GuardedBy annotation, not JCIP's or Javax's)
+            // for further subtype checking.
+
+            return super.isSubtype(rhsIsGuardedBy ? GUARDEDBY : rhs, lhsIsGuardedBy ? GUARDEDBY : lhs);
+        }
+    }
+
+    // Returns the list of expressions specified in the @Holding (or equivalent) annotation on a method.
+    // Returns an empty list if such an annotation is not present.
+    protected List<String> methodHolding(ExecutableElement element) {
+        AnnotationMirror holding = getDeclAnnotation(element, checkerHoldingClass);
+        AnnotationMirror guardedBy
+            = getDeclAnnotation(element, jcipGuardedByClass);
+        AnnotationMirror guardedByJavax
+            = getDeclAnnotation(element, javaxGuardedByClass);
+
+        if (holding == null && guardedBy == null && guardedByJavax == null)
+            return Collections.emptyList();
+
+        List<String> locks = new ArrayList<String>();
+
+        if (holding != null) {
+            List<String> holdingValue = AnnotationUtils.getElementValueArray(holding, "value", String.class, false);
+            locks.addAll(holdingValue);
+        }
+        if (guardedBy != null) {
+            String guardedByValue = AnnotationUtils.getElementValue(guardedBy, "value", String.class, false);
+            locks.add(guardedByValue);
+        }
+        if (guardedByJavax != null) {
+            String guardedByValue = AnnotationUtils.getElementValue(guardedByJavax, "value", String.class, false);
+            locks.add(guardedByValue);
+        }
+
+        return locks;
+    }
+
+    // The side effect annotations processed by the Lock Checker,
+    // in order of increasingly strong guarantees.
+    enum SideEffectAnnotation {
+        MAYRELEASELOCKS,
+        RELEASESNOLOCKS,
+        LOCKINGFREE,
+        SIDEEFFECTFREE,
+        PURE
+    }
+
+    // Indicates which side effect annotation is present on the given method.
+    // If more than one annotation is present, this method issues an error (if issueErrorIfMoreThanOnePresent is true)
+    // and returns the annotation providing the weakest guarantee.
+    // If no annotation is present, return RELEASESNOLOCKS as the default, and MAYRELEASELOCKS
+    // as the default for unannotated code.
+    SideEffectAnnotation methodSideEffectAnnotation(Element element, boolean issueErrorIfMoreThanOnePresent) {
```

Comment:
There is one such call, in LockVisitor.visitMethod. This parameter allows errors to be issued when visiting a method declaration but not when visiting every method call.